django/db/models/query.py

```diff
@@ -815,6 +815,21 @@ def complex_filter(self, filter_obj):
         else:
             return self._filter_or_exclude(None, **filter_obj)
 
+    def _combinator_query(self, other_qs, combinator):
+        combined_qs = self.__class__(model=self.model, using=self._db, hints=self._hints)
+        combined_qs.query.combinator_querysets = (self.all(), other_qs.all())
```

Comment:
Why are you copying the `QuerySet`s? Shouldn't be necessary as all their attributes are immutable except outside of other operations, and the result cache doesn't seem to affect their use in the combined qs.